functorch/_src/aot_autograd.py

```diff
@@ -1488,18 +1488,30 @@ def create_aot_dispatcher_function(
     if config.use_dynamic_shapes:
         assert config.use_fake_tensor, "Dynamic shapes only works with fake tensor"
 
-    shape_env = ShapeEnv() if config.use_dynamic_shapes else None
-    fake_mode = FakeTensorMode(shape_env=shape_env) if config.use_fake_tensor else nullcontext()
+    # Check flat_args to see if they're already fake.  If so, use that fake
+    # mode instead.
+
+    for x in flat_args:
+        if isinstance(x, FakeTensor):
+            fake_mode = x.fake_mode
+            break
+    else:
+        shape_env = ShapeEnv() if config.use_dynamic_shapes else None
+        fake_mode = FakeTensorMode(shape_env=shape_env) if config.use_fake_tensor else nullcontext()
+
     cross_ref = CrossRefFakeMode() if config.debug_fake_cross_ref else nullcontext()
     python_dispatcher_mode = enable_python_dispatcher() if config.use_dynamic_shapes else nullcontext()
 
     with torch.autograd.set_multithreading_enabled(False), preserve_rng_state(), cross_ref, fake_mode, python_dispatcher_mode:
 
         def process_inputs(flat_args):
-            if config.use_fake_tensor:
+            if config.use_fake_tensor or isinstance(fake_mode, FakeTensorMode):
                 def convert(idx, x):
                     if not isinstance(x, torch.Tensor):
                         return x
+                    if isinstance(x, FakeTensor):
+                        assert x.fake_mode is fake_mode
+                        return x
                     if idx < aot_config.num_params_buffers and config.static_weight_shapes:
                         return fake_mode.from_tensor(x, static_shapes=True)
                     return fake_mode.from_tensor(x, static_shapes=False)
```

Comment:
Ive gone back and forth on the signal here.

1) Dynamo sets `config.use_fake_tensor` correctly
2) We check `config.use_fake_tensor` in other place
3) You should never be in a state where you are in `config.use_fake_tensor == False` and getting a fake mode

So maybe instead we either:

A) Assert that we are in fake config mode if we see a fake tensor in inputs on L1495 and use the presence of fake_mode to be analogous to config.use_fake_tensor (since you can only get one via inputs in config.use_fake_tensor or by making your own on L1500, also within config)

B) Or drop the isinstance(fake_mode) checks and assume that if we have the config set, we have the fake mode, and guard that with an assert.

Reply:
I don't think this is worth litigating much, because I think we should delete this config option (e.g., https://github.com/pytorch/pytorch/pull/89663 ).